tendermint/src/rpc/endpoint/abci_info.rs

```diff
@@ -27,41 +27,58 @@ impl rpc::Response for Response {}
 
 /// ABCI information
 #[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(default)]
 pub struct AbciInfo {
     /// Name of the application
     pub data: String,
 
     /// Version
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub version: Option<String>,
 
     /// Last block height
-    pub last_block_height: block::Height,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub last_block_height: Option<block::Height>,
 
     /// Last app hash for the block
     #[serde(
         serialize_with = "serialize_app_hash",
-        deserialize_with = "parse_app_hash"
+        deserialize_with = "parse_app_hash",
+        skip_serializing_if = "Option::is_none"
     )]
-    pub last_block_app_hash: Hash,
+    pub last_block_app_hash: Option<Hash>,
 }
 
 /// Parse Base64-encoded app hash
-pub(crate) fn parse_app_hash<'de, D>(deserializer: D) -> Result<Hash, D::Error>
+pub(crate) fn parse_app_hash<'de, D>(deserializer: D) -> Result<Option<Hash>, D::Error>
 where
     D: Deserializer<'de>,
 {
     let bytes = base64::decode(String::deserialize(deserializer)?.as_bytes())
         .map_err(|e| D::Error::custom(format!("{}", e)))?;
 
-    Hash::new(hash::Algorithm::Sha256, &bytes).map_err(|e| D::Error::custom(format!("{}", e)))
+    Hash::new(hash::Algorithm::Sha256, &bytes)
+        .map(Some)
+        .map_err(|e| D::Error::custom(format!("{}", e)))
 }
 
 /// Serialize Base64-encoded app hash
-pub(crate) fn serialize_app_hash<S>(hash: &Hash, serializer: S) -> Result<S::Ok, S::Error>
+pub(crate) fn serialize_app_hash<S>(hash: &Option<Hash>, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
 {
-    String::from_utf8(base64::encode(hash.as_bytes()))
+    String::from_utf8(base64::encode(hash.unwrap().as_bytes()))
         .unwrap()
         .serialize(serializer)
 }
+
+impl Default for AbciInfo {
+    fn default() -> Self {
+        AbciInfo {
+            data: "".to_string(),
+            version: None,
+            last_block_height: None,
+            last_block_app_hash: None,
+        }
+    }
+}
```

Comment:
Add a comment why `Default` is needed. (My best guess is because is that we want to be able to skip over fields but I might be wrong).

Reply:
Yes. Normally this is not needed but unfortunately the serialization implementation of `last_block_height` and `last_block_app_hash` is custom, hence they don't have any defaults for the case when they are missing from a deserialization.

Reply:
Yeah, it's probably worth capturing this reason in the code directly.

Reply:
Yup, added it.